### ISSUE TITLE
Git push hangs on "Pushing to registry"

### ISSUE THREAD

**Issue by joshma** [deis/deis#5121]
Hi there, I'm testing out a deis setup, and `git push deis master` is hanging:
```
Step 7 : CMD node bot.js
 ---> Using cache
 ---> 2625f852a64b
Successfully built 2625f852a64b
Pushing to registry
# Hangs here for a long time
fatal: The remote end hung up unexpectedly
fatal: The remote end hung up unexpectedly
```

Some notes on my setup:
- Using kops in AWS, enabled CNI (calico). Using an existing VPC but dedicated subnets
- When I set up deis, I changed `values.yml` to `use_cni: true` and `registry_proxy_bind_addr: "127.0.0.1:5555"
- Checked `kubectl --namespace=deis get pods` and all pods are running
- The application in question uses a Dockerfile.

Is there any way I can troubleshoot this? Earlier I got a different error when I didn't configure CNI correctly, but now it's just mysteriously hanging. As far as I understand, the default setup uses a private Docker registry, so I don't need to configure ECR, etc - is that correct? Is there a kubernetes registry that I need to configure?

**Comment by Joshua-Anderson** [comment]
This error message means that something in the system timed out. Often, the ELB is the cause. Have you raised your ELB timeout?

**Comment by joshma** [comment]
Yes - my timeout is set to 20 minutes, although I didn't have to do that explicitly - it was already done for me. Is it expected that pushing to registry takes so long?

**Comment by Joshua-Anderson** [comment]
Since these are containers with a full heroku stack on it, the initial container push is very large so it could take some time.
